### PR TITLE
Added GlobalRegionScheduler#getExecutor and RegionScheduler#getExecutor

### PR DESCRIPTION
**Pull request by jdkeke142**
I added GlobalRegionScheduler#getExecutor and RegionScheduler#getExecutor to get respectively an executor of the global region and to get an executor of a corresponding region at a location. I find that these methods were useful in regular paper when working with CompletableFuture, I think other developers too. This PR aims to reintroduce them

### CODE REVIEW

**Comment by Machine-Maker** [comment]
I think the method names should just be getExecutor(). They are on interfaces with the names Region and Global, it doesn't need to be repeated in the method name.

File: patches/api/0006-Added-GlobalRegionScheduler-getGlobalThreadExecutor-.patch
Comment (kashike) [inline patches/api/0006-Added-GlobalRegionScheduler-getGlobalThreadExecutor-.patch:1]: Patch could use a shorter name.

Comment (jdkeke142) [reply patches/api/0006-Added-GlobalRegionScheduler-getGlobalThreadExecutor-.patch:1]: It should be better now

Comment (kashike) [reply patches/api/0006-Added-GlobalRegionScheduler-getGlobalThreadExecutor-.patch:1]: Sorry, to clarify, you'll have to edit the commit summary for the patch files to retain shorter names - you can shove the longer description into the commit description.

Comment (jdkeke142) [reply patches/api/0006-Added-GlobalRegionScheduler-getGlobalThreadExecutor-.patch:1]: Sorry it's me not being familiar with the patch system. Does it look better now?

**Review by Yomamaeatstoes — COMMENTED** [review]
Looks good

**Comment by Spottedleaf** [comment]
I don't want to encourage usage of CompletableFuture, since they do not follow reasonable error handling. I've learned this the hard way from Mojang's code, and I don't expect plugins to do a better job.

**Pull request closed without merging**